week-9/Homework/mandatory/1-practice/2-code-reading.md

```diff
@@ -61,3 +64,6 @@ console.log(y);
 ```
 
 What will be the output of this code. Explain your answer in 50 words or less.
+
+First output will be 9 because we only console.log the variable x witch is equal with 9
+Second output will an object with a key x witch has a value of 9
```

Comment:
The output of the first scenario will be 9.
The output of the second scenario will be 10.

In scenario 1, we pass in a number so we are passing by value. Changing the argument passed into the function will not change the variable outside of the function.

In scenario 2, we pass in an object so we are passing by reference. This means that if you make changes to the argument if will affect the object in the variable outside the function.

https://medium.com/nodesimplified/javascript-pass-by-value-and-pass-by-reference-in-javascript-fcf10305aa9c